components/places/src/storage/bookmarks/json_tree.rs

```diff
@@ -0,0 +1,808 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// This supports inserting and fetching an entire bookmark tree via JSON
+// compatible data structures.
+// It's currently used only by tests, examples and our utilities for importing
+// from a desktop JSON exports.
+//
+// None of our "real" consumers currently require JSON compatibility, so try
+// and avoid using this if you can!
+// (We could possibly put this behind a feature flag?)
```

Comment:
I would vote for this path. Feels like in the future it might be all too easy to want to just via JSON it and maybe making it more explicit will deter from that, but don't feel super strongly -- these comments feel enough for any potential future consumer.